src/exchanges/ssr.test.ts

```diff
@@ -0,0 +1,91 @@
+import { makeSubject, pipe, map, publish, forEach, Subject } from 'wonka';
+
+import { Client } from '../client';
+import { queryOperation, queryResponse } from '../test-utils';
+import { ExchangeIO, Operation } from '../types';
+import { ssrExchange } from './ssr';
+
+let forward: ExchangeIO;
+let exchangeInput;
+let client: Client;
+let input: Subject<Operation>;
+let output;
+
+beforeEach(() => {
+  input = makeSubject<Operation>();
+  output = jest.fn();
+  forward = ops$ =>
+    pipe(
+      ops$,
+      map(output)
+    );
+  client = { suspense: true } as any;
+  exchangeInput = { forward, client };
+});
+
+it('caches query results correctly', () => {
+  output.mockReturnValueOnce(queryResponse);
+
+  const ssr = ssrExchange();
+  const [ops$, next] = input;
+  const exchange = ssr(exchangeInput)(ops$);
+
+  publish(exchange);
+  next(queryOperation);
+
+  const data = ssr.extractData();
+  expect(Object.keys(data)).toEqual(['' + queryOperation.key]);
```

Comment:
Is this `'' + queryOperation.key` purely for string coercion? I'd probably just use a template literal or `.toString` to keep intent clear.

Reply:
It's just for string coercion and I prefer `'' + x` over the template string variant at least for testing tbh 😅